### ISSUE TITLE
Feature Request: Saving data for recache

### ISSUE THREAD

**Issue by Gordon-TheTurtle** [objectcomputing/liquibook#27]
It would be great to add serialization abilities in liquibook for ease of system restart.

**Comment by enewhuis** [comment]
:D yes, great idea!  This one is at the top of my personal list.  I've hacked partial support for that on a private branch that is storage specific, but perhaps there is a better generic way to do this.  I was thinking Liquibook ought to have both a checkpointing and journal recovery feature so that it could be restarted, load checkpoint state, and then play queued incoming orders.

**Comment by Gordon-TheTurtle** [comment]
Yeah, that is a use case every one using liquibook would have for recovery purposes.
I propose to use boost serialization abilities as boost is already integrated.
And then let the user use these abilities as per their needs to give a fine-grained control.

It will also enable the user to save any additional data that they might have themselves, in a single call to save latency, and show a consistent state in case of failure in saving.
